opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -992,29 +993,41 @@ namespace Opm {
         {
             using namespace Opm::AutoDiffGrid;
             const int nc = numCells(grid_);
-            //const ADB pv_mult = poroMult(pressure);
-            const auto& pv = geo_.poreVolume();
             const int maxnp = Opm::BlackoilPhases::MaxNumPhases;
 
             for (int i = 0; i<FIPData::fipValues; i++) {
                 fip_.fip[i].resize(nc,0.0);
             }
 
-            for (int c = 0; c < nc; ++c) {
-                const auto& intQuants = *ebosSimulator_.model().cachedIntensiveQuantities(c, /*timeIdx=*/0);
+            ElementContext elemCtx(ebosSimulator_);
+            auto elemIt = elemCtx.gridView().template begin</*codim=*/0>();
+            const auto& elemEndIt = elemCtx.gridView().template end</*codim=*/0>();
+            for (; elemIt != elemEndIt; ++elemIt) {
+                const auto& elem = *elemIt;
+                if (elem.partitionType() != Dune::InteriorEntity)
+                    continue;
+
+                elemCtx.updatePrimaryStencil(elem);
+                elemCtx.updatePrimaryIntensiveQuantities(/*timeIdx=*/0);
+
+                const unsigned cellIdx = elemCtx.globalSpaceIndex(/*spaceIdx=*/0, /*timeIdx=*/0);
+                const auto& intQuants = elemCtx.intensiveQuantities(/*spaceIdx=*/0, /*timeIdx=*/0);
                 const auto& fs = intQuants.fluidState();
+                const double pv =
+                    ebosSimulator_.model().dofTotalVolume(cellIdx)
+                    * ebosSimulator_.problem().porosity(cellIdx);
```

Comment:
These are a lot of new lines of code. As far as I can judge the only change is that you now calculate the pore volume from ebos properties instead of taking the one stored in the properties. Is this the fix, i.e. are the values in `geo_` wrong. If not, what is the reason to obfuscate it with DUNE/ebos lingo and why is it even part of this PR?